src/DataStructures/Matrix.hpp

```diff
@@ -66,3 +77,13 @@ bool operator==(const Matrix& lhs, const Matrix& rhs);
 
 /// Inequivalence operator for Matrix
 bool operator!=(const Matrix& lhs, const Matrix& rhs);
+
+/// Matrix multiplication
+Matrix operator*(const Matrix& lhs, const Matrix& rhs);
+
+// I just put these here for now, too.
+DataVector dot_product(const Matrix& lhs, const DataVector& rhs);
```

Comment:
Can we stop calling things that aren't dot products "dot product"?

Reply:
how would you like to call it?

Reply:
I'd be fine with `multiply`

Reply:
That's just the `*` operator written in words :D

Reply:
Yes, but we don't want people to be able to use `*`